core/backend/src/IModelTransformer.ts

```diff
@@ -766,11 +766,19 @@ export class IModelTransformer extends IModelExportHandler {
     return Semver.gt(`${schemaKey.version.read}.${schemaKey.version.write}.${schemaKey.version.minor}`, Schema.toSemverString(versionInTarget));
   }
 
+  private _hasNativelyExportedAllSchemas = false;
+
   /** Override of [IModelExportHandler.onExportSchema]($backend) that serializes a schema to disk for [[processSchemas]] to import into
    * the target iModel when it is exported from the source iModel. */
-  protected override async onExportSchema(schema: ECSchemaMetaData.Schema): Promise<void> {
-    const schemaPath = path.join(this._schemaExportDir, `${schema.fullName}.ecschema.xml`);
-    IModelJsFs.writeFileSync(schemaPath, await schema.toXmlString());
+  protected override async onExportSchema(_schema: ECSchemaMetaData.Schema): Promise<void> {
+    // HACK: a bug in the native deserializer means we can't yet always read schemas serialized from this end

```

Comment:
Is there a fix in the works for the native side?

Reply:
yes I submitted a native PR and will submit another one here reverting this change with the test passing.